test/models/test_albef.py

```diff
@@ -4,113 +4,220 @@
+
+@pytest.fixture(autouse=True)
+def albef_model_output(albef_model):
+    image = torch.randn(2, 3, 4, 4)
+    text = torch.randint(10, (2, 2))
+    text_atts = Tensor([[1, 1], [1, 0]])
+    return albef_model(image, text, text_atts)
+
+
+def test_albef_image_embeddings(albef_model_output):
+    expected = Tensor(
+        [
+            [[1.401580, -0.537510, -0.864071], [1.378901, -0.417473, -0.961429]],
+            [[1.413948, -0.729806, -0.684143], [-1.313033, 1.111438, 0.201595]],
+        ]
+    )
+    assert_expected(albef_model_output.image_embeddings, expected, rtol=0, atol=1e-4)
+
+
+def test_albef_image_embeddings_momentum(albef_model_output):
+    expected = Tensor(
+        [
+            [[1.401580, -0.537510, -0.864070], [1.378902, -0.417473, -0.961429]],
+            [[1.413949, -0.729807, -0.684141], [-1.313033, 1.111438, 0.201595]],
+        ]
+    )
+    assert_expected(albef_model_output.image_embeddings_m, expected, rtol=0, atol=1e-4)
+
+
+def test_albef_text_embeddings(albef_model_output):
+    expected = Tensor(
+        [
+            [[-1.372420, 0.981756, 0.390664], [1.305102, -1.124284, -0.180818]],
+            [[-1.320019, 0.220507, 1.099512], [1.366452, -0.998831, -0.367621]],
+        ]
+    )
+    assert_expected(albef_model_output.text_embeddings, expected, rtol=0, atol=1e-4)
+
+
+def test_albef_vl_embeddings(albef_model_output):
+    expected = Tensor(
+        [
+            [-1.398094, 0.883438, 0.514656],
+            [-1.115577, 1.310528, -0.194951],
+            [-0.706023, 1.414213, -0.708190],
+            [-1.402520, 0.544084, 0.858435],
+            [-1.402520, 0.544084, 0.858435],
+            [-0.706023, 1.414213, -0.708190],
+        ]
+    )
+    assert_expected(albef_model_output.vl_embeddings, expected, rtol=0, atol=1e-4)
+
+
+def test_copy_params_momentum_models(dummy_albef_model):
+    dummy_albef_model.models_m = [nn.Linear(3, 2) for _ in range(5)]
+    dummy_albef_model._copy_params_momentum_models()
+    for model, model_m in zip(dummy_albef_model.models, dummy_albef_model.models_m):
+        for param, param_m in zip(model.parameters(), model_m.parameters()):
+            assert_expected(param, param_m, rtol=0, atol=1e-4)
+            assert not param_m.requires_grad
+
+
+def test_dequeue_and_enqueue(dummy_albef_model):
+    image_feat_m = torch.randn(2, 2)
+    text_feat_m = torch.randn(2, 2)
+    dummy_albef_model._dequeue_and_enqueue(image_feat_m, text_feat_m)
+    assert_expected(
+        dummy_albef_model.image_queue[:, 0:2], image_feat_m.T, rtol=0, atol=1e-4
+    )
+    assert_expected(
+        dummy_albef_model.text_queue[:, 0:2], text_feat_m.T, rtol=0, atol=1e-4
+    )
+
+
+def test_momentum_update(dummy_albef_model):
+    init_weight = Tensor([[1, 2, 3], [4, 5, 6]])
+    init_weight_m = Tensor([[6, 5, 4], [3, 2, 1]])
+    dummy_albef_model.models[0].weight = nn.Parameter(init_weight)
+    dummy_albef_model.models_m[0].weight = nn.Parameter(init_weight_m)
```

Comment:
Kind of a nit, but this indexing to get the vision encoder out of the ALBEFModel class makes me think that a list is probably not the best data structure for `self.models` and `self.models_m`...

Reply:
All models can still be accessed directly (e.g. `dummy_albef_model.vision_encoder`). The two lists are used to iterate through all pairs of `(model, model_m)` (which happens in `_copy_params_momentum_models` and `_momentum_update`). Should I rename them to `_models` and `_models_m`? Or try other data structures (dict `{'vision_encoder': (vision_encoder, vision_encoder_m), ...}`?

Reply:
@katrina433 good point about just accessing via `.vision_encoder`. Personally I would do that here just to be more explicit about what you're modifying. I think @YosuaMichael's suggestion makes sense, feel free to address this in a separate PR though